.devcontainer/Dockerfile

```diff
@@ -333,40 +344,78 @@ RUN apt-key add /tmp/jfrog.pub && \
 
 #### protos
-ARG PYTHON_LIB=/usr/local/lib/python3.8
+ARG GEN_DIR=lib/python${PYTHON_VERSION}/gen
 
 COPY /protos/ ${MAGMA_ROOT}/protos/
 COPY /lte/protos/ ${MAGMA_ROOT}/lte/protos/
 COPY /orc8r/protos/ ${MAGMA_ROOT}/orc8r/protos/
 COPY /feg/protos/ ${MAGMA_ROOT}/feg/protos/
 COPY /dp/protos/ ${MAGMA_ROOT}/dp/protos/
-WORKDIR $MAGMA_ROOT
-RUN pip install mypy-protobuf && \
-    mkdir $PYTHON_LIB/gen && \
+WORKDIR ${MAGMA_ROOT}
+RUN ${PYTHON_VENV_EXECUTABLE} -m pip install --no-cache-dir "mypy-protobuf==2.4" && \
+    mkdir ${PYTHON_VENV}/${GEN_DIR} && \
     for PROTO_SRC in orc8r lte feg dp; \
     do \
-    python protos/gen_protos.py ${PROTO_SRC}/protos ${MAGMA_ROOT},orc8r/protos/prometheus ${MAGMA_ROOT} $PYTHON_LIB/gen && \
-    python protos/gen_prometheus_proto.py ${MAGMA_ROOT} $PYTHON_LIB/gen; \
+    ${PYTHON_VENV_EXECUTABLE} protos/gen_protos.py ${PROTO_SRC}/protos ${MAGMA_ROOT},orc8r/protos/prometheus ${MAGMA_ROOT} ${PYTHON_VENV}/${GEN_DIR} && \
+    ${PYTHON_VENV_EXECUTABLE} protos/gen_prometheus_proto.py ${MAGMA_ROOT} ${PYTHON_VENV}/${GEN_DIR}; \
     done && \
-    echo "$PYTHON_LIB/gen" > $PYTHON_LIB/dist-packages/magma_gen.pth
+    echo "${PYTHON_VENV}/${GEN_DIR}" > ${PYTHON_VENV}/lib/python${PYTHON_VERSION}/site-packages/magma_gen.pth
+
+### swagger
+ENV SWAGGER_CODEGEN_JAR=/var/tmp/codegen/modules/swagger-codegen-cli/target/swagger-codegen-cli.jar
+ARG MVN_VERSION=3.5.4
+ARG CODEGEN_VERSION=v2.2.3
+
+WORKDIR /var/tmp/
+
+RUN wget http://mirrors.ibiblio.org/apache/maven/maven-3/${MVN_VERSION}/binaries/apache-maven-${MVN_VERSION}-bin.tar.gz && \
+    tar --extract --verbose --file apache-maven-${MVN_VERSION}-bin.tar.gz
+
+RUN git clone --branch ${CODEGEN_VERSION} https://github.com/swagger-api/swagger-codegen /var/tmp/codegen
+
+WORKDIR /var/tmp/codegen
+RUN ../apache-maven-${MVN_VERSION}/bin/mvn clean package -DskipTests
+
+COPY lte/swagger/*.yml /tmp/lte/swagger/
+COPY orc8r/swagger/*.yml /tmp/orc8r/swagger/
+RUN for SWAGGER_SRC in lte orc8r; \
+    do \
+    mkdir -p ${PYTHON_VENV}/${GEN_DIR}/${SWAGGER_SRC}/swagger/specs && \
+    # Copy swagger specs over to the build directory,
+    # so that eventd can access them at runtime
+    cp /tmp/${SWAGGER_SRC}/swagger/* ${PYTHON_VENV}/${GEN_DIR}/${SWAGGER_SRC}/swagger/specs && \
+    # Generate the files
+    ls ${PYTHON_VENV}/${GEN_DIR}/${SWAGGER_SRC}/swagger/specs \
+    | grep -e ".*\.yml" \
+    | xargs -t -I% /usr/bin/java -jar ${SWAGGER_CODEGEN_JAR} generate \
+    -i ${PYTHON_VENV}/${GEN_DIR}/${SWAGGER_SRC}/swagger/specs/% \
+    -o ${PYTHON_VENV}/${GEN_DIR}/${SWAGGER_SRC}/swagger \
+    -l python \
+    -Dmodels && \
+    # Flatten and clean up directory
+    mv ${PYTHON_VENV}/${GEN_DIR}/${SWAGGER_SRC}/swagger/swagger_client/* ${PYTHON_VENV}/${GEN_DIR}/${SWAGGER_SRC}/swagger/ && \
+    rmdir ${PYTHON_VENV}/${GEN_DIR}/${SWAGGER_SRC}/swagger/swagger_client && \
+    rm -r ${PYTHON_VENV}/${GEN_DIR}/${SWAGGER_SRC}/swagger/test && \
+    rm -r /tmp/${SWAGGER_SRC}/swagger/; \
```

Comment:
We could even remove the '/swagger` subdir everywhere in `/tmp`, because I do not see any meaningful usage of it.

Reply:
I chose a slightly different approach: through out the `cp` statement that used to be in L403 within the loop. Since we have a docker `COPY` statement for every item in the loop anyways, I just do the copying there. That gets rid of the /tmp/... directories entirely and also of the `mkdir` statement.

I checked: the swagger generation creates the following directories `swagger_client/models` `docs` and `test`. Apparently we delete `test` and move `models` one level up (plus any other files in `swagger_client`, I only checked for the orc8r and there `swagger_client` was empty except for `models`)